File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/XposedCollector.java
package cn.tongdun.mobrisk.core.collectors;


/**
 * @author collam
 * @date 2023/2/20
 */
public class XposedCollector {
    public boolean findXposed() {
        try {
            Class.forName("de.robv.android.xposed.XposedBridge",true,ClassLoader.getSystemClassLoader());
            return true;
        }
        catch (Exception ignore){
        }
        return false;
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/XposedCollector.kt
package cn.tongdun.mobrisk.core.collectors

/**
 * @description:
 * @author: wuzuchang
 * @date: 2023/5/15
 */
interface XposedInterface {
    fun findXposed(): Boolean
}

class XposedCollector : XposedInterface {
    override fun findXposed(): Boolean {
        val xposedClasses = arrayOf(
            "de.robv.android.xposed.XposedBridge",
            "de.robv.android.xposed.IXposedHookLoadPackage",
            "de.robv.android.xposed.DexposedBridge"
        )
        for (className in xposedClasses) {
            try {
                Class.forName(className, true, ClassLoader.getSystemClassLoader())
                return true
            } catch (ignore: Exception) {
            }
        }
        return false
    }
}